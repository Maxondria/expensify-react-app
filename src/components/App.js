import React from "react";
import AppRouter from "../router/AppRouter";
import Header from "./Header";
import { createBrowserHistory } from "history";
import { Router } from "react-router-dom";

export const history = createBrowserHistory();

const App = () => (
  <Router history={history}>
    <Header />
    <AppRouter />
  </Router>
);

export default App;
