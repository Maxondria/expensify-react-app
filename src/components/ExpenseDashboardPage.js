import React from "react";

const ExpenseDashboardPage = () => <h1>Expense Dashboard</h1>;

export default ExpenseDashboardPage;
