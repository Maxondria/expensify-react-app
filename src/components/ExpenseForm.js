import React from "react";
import moment from "moment";
import "react-dates/initialize";
import "react-dates/lib/css/_datepicker.css";
import { SingleDatePicker } from "react-dates";

export default class ExpenseForm extends React.Component {
  state = {
    amount: "",
    description: "",
    note: "",
    calenderFocused: false,
    date: moment()
  };

  OnChange = ({ target: { name, value } }) => {
    this.setState(prevState => ({
      ...prevState,
      [name]:
        name === "amount"
          ? value.match(/^\d+(\.\d{0,2})?$/)
            ? value
            : ""
          : value
    }));
  };

  OnSave = e => {
    e.preventDefault();
    console.log(this.state);
  };

  OnFocusChange = ({ focused }) => this.setState({ calenderFocused: focused });

  OnDateChange = date => this.setState({ date });

  render() {
    return (
      <div>
        <form>
          <input
            type="text"
            name="description"
            value={this.state.description}
            placeholder="Description"
            onChange={this.OnChange}
            autoFocus
          />
          <input
            type="text"
            name="amount"
            value={this.state.amount}
            onChange={this.OnChange}
            placeholder="Amount"
          />
           {/* eslint-disable react/prop-types */}
          <SingleDatePicker
            date={this.state.date}
            onDateChange={this.OnDateChange}
            focused={this.state.calenderFocused}
            onFocusChange={this.OnFocusChange}
            numberOfMonths={1}
            isOutsideRange={() => false}
            id="CreatedAt"
          />
          <textarea
            cols="20"
            rows="5"
            name="note"
            placeholder="Add A Note"
            onChange={this.OnChange}
            value={this.state.note}
          />
          <button onClick={this.OnSave}>Save Expense</button>
        </form>
      </div>
    );
  }
}
