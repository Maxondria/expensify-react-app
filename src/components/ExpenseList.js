import React from "react";
import { connect } from "react-redux";
import ExpenseListItem from "./ExpenseListItem";

const ExpenseList = props => (
  <div>
    <h1>This is my Expenses List</h1>
    {props.expenses.map(expense => (
      <ExpenseListItem {...expense} />
    ))}
  </div>
);

const mapStateToProps = ({ expenses, filters }) => ({
  expenses,
  filters
});

export default connect(mapStateToProps)(ExpenseList);
