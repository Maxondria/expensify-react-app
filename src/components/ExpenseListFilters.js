import React from "react";
import { connect } from "react-redux";
import {
  setTextFilter,
  sortByAmount,
  sortByDate
} from "../redux/actions/filters";

const ExpenseListFilters = props => (
  <div>
    <input
      type="text"
      value={props.filters.text}
      onChange={e => props.setTextFilter(e.target.value)}
    />

    <select
      name="sort_by"
      id="sort_by"
      value={props.filters.sortBy}
      onChange={e => {
        return e.target.value === "date"
          ? props.sortByDate()
          : props.sortByAmount();
      }}
    >
      <option value="date">Date</option>
      <option value="amount">Amount</option>
    </select>
  </div>
);

const mapStateToProps = ({ filters }) => ({
  filters
});

const mapDispatchToProps = {
  setTextFilter,
  sortByAmount,
  sortByDate
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ExpenseListFilters);
