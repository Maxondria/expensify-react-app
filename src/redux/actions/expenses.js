import actionTypes from "../constants";
import uuid from "uuid";

const addExpense = ({
  description = "",
  note = "",
  amount = 0,
  createdAt = 0
} = {}) => ({
  type: actionTypes.ADD_EXPENSE,
  expense: {
    id: uuid(),
    description,
    note,
    amount,
    createdAt
  }
});

const removeExpense = (id) => ({
  type: actionTypes.REMOVE_EXPENSE,
  id
});

const editExpense = (id, expense) => ({
  type: actionTypes.EDIT_EXPENSE,
  expense
});

export { addExpense, removeExpense, editExpense };
