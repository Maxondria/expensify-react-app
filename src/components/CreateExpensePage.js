import React from "react";
import ExpenseForm from "./ExpenseForm";
import { connect } from "react-redux";
import { addExpense } from "../redux/actions/expenses";

export class CreateExpensePage extends React.Component {
  onSubmit = expense => {
    this.props.addExpense(expense);
    this.props.history.push("/");
  };

  render() {
    return (
      <div>
        <h1>Add An Expense</h1>
        <ExpenseForm onSubmit={this.onSubmit} />
      </div>
    );
  }
}

const mapDispatchToProps = {
  addExpense
};

export default connect(
  undefined,
  mapDispatchToProps
)(CreateExpensePage);
