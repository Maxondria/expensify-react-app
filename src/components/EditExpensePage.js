import React from "react";
import { connect } from "react-redux";
import ExpenseForm from "./ExpenseForm";
import { editExpense, startRemoveExpense } from "../redux/actions/expenses";

export class EditExpensePage extends React.Component {
  id = this.props.match.params.id;

  onSubmit = ({ note, description, amount, date: createdAt }) => {
    this.props.editExpense(this.id, {
      note,
      description,
      amount,
      createdAt
    });
    this.props.history.push("/");
  };

  onRemoveExpense = () =>
    this.props.startRemoveExpense(this.id) && this.props.history.push("/");

  render() {
    return (
      <div>
        {this.props.expense.length === 0 ? (
          <p>Expense Not Found</p>
        ) : (
          <>
            <ExpenseForm
              expense={this.props.expense}
              onSubmit={this.onSubmit}
            />
            <button onClick={this.onRemoveExpense}>Remove</button>
          </>
        )}
      </div>
    );
  }
}

const mapStateToProps = ({ expenses }, props) => ({
  expense: expenses.find(expense => expense.id === props.match.params.id)
});

const mapDispatchToProps = {
  editExpense,
  startRemoveExpense
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(EditExpensePage);
