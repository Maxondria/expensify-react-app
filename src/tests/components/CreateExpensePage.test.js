import React from "react";
import { shallow } from "enzyme";
import { CreateExpensePage } from "../../components/CreateExpensePage";
import expenses from "../redux/fixtures/expenses.fixture";

const props = {
  addExpense: jest.fn(),
  history: {
    push: jest.fn()
  }
};

it("should render CreateExpensePage correctly", function() {
  const wrapper = shallow(<CreateExpensePage {...props} />);

  expect(wrapper).toMatchSnapshot();
});

it("should handle onSubmit()", function() {
  const wrapper = shallow(<CreateExpensePage {...props} />);

  wrapper.find("ExpenseForm").prop("onSubmit")(expenses[0]);
  expect(props.addExpense).toHaveBeenLastCalledWith(expenses[0]);
  expect(props.history.push).toHaveBeenLastCalledWith("/");
});
