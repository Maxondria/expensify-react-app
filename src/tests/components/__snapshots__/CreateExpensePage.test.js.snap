// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CreateExpensePage correctly 1`] = `
<div>
  <h1>
    Add An Expense
  </h1>
  <ExpenseForm
    onSubmit={[Function]}
  />
</div>
`;
